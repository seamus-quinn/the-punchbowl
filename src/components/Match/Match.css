.track {
  display: grid;
  grid-template-columns: 300px 1fr;
  height: 600px;
  margin: 0 3%;
}

.keyword-card {
  width: 300px;
  color: #fff;  
  background: #181921;
}

.article-container {
  display: flex;
  flex-direction: row;
  overflow-x: auto;
}

.article-card {
  flex: 0 0 auto;
  width: 400px;
  height: 100%;
  color: #fff;
  background: #181921;
  margin: 0 8px;
  background: #fff;
}

.image {
  width: 100%;
  height: 220px;
}

.card-info {
  display: flex;
  flex-direction: column;
  align-items: space-between;
}

.article-title {
  color: #181921;
  font-size: 24px;
  text-align: left;
  padding: 0 4%;
}

.article-description {
  color: #181921;
  text-align: left;
  padding: 0 4%;
}

.link-to-article {

}

