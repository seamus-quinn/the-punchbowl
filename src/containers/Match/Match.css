.track {
  display: grid;
  grid-template-columns: 300px 1fr;
  height: 600px;
  margin: 0 3%;
}

.keyword-card {
  width: 300px;
  color: #fff;  
  background: rgba(255, 255, 255, 0.2);
  position: relative;
}

.keyword-text {
  position: absolute;
  margin-left: 20px
}

.keyword-list {
  font-size: 11px;
  text-align: left;
  padding-left: 25px;
}

.match-delete-button {
  position: absolute;
  height: 50px;
  width: 50px;
  outline: none;
  border: none;
  background: none;
  padding-top: 10px;
  right: 10px;
}

.article-counter {
  position: absolute;
  bottom: 10px;
  right: 20px;
}

.article-container {
  display: flex;
  flex-direction: row;
  overflow-x: auto;
  margin-left: 20px;
}

.article-card {
  flex: 0 0 auto;
  width: 400px;
  height: 100%;
  color: #fff;
  margin: 0 8px;
  background: #fff;
}

.image-container {
  height: 220px;
}

.image {
  width: 100%;
  height: 100%;
}

.card-info {
  display: flex;
  flex-direction: column;
  height: 380px;
  align-items: space-between;
  position: relative;
  margin: 0;
}

.article-title {
  color: #181921;
  font-size: 26px;
  font-weight: bold;
  text-align: left;
  padding: 0 5%;
}

.article-description, 
.article-source {
  color: #181921;
  text-align: left;
  padding: 0 5%;
}

.link-to-article {
  position: absolute;
  bottom: 20px;
  right: 20px;
  text-decoration: none;
  color: #15d1df
}

.link-to-article:visited {
  color: #c782ff
}



