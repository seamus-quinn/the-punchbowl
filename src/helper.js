export const makeWordsArr = (arr) => {
  const words = arr.reduce((acc, article) => {
    if (article.description) {
      const title = article.title.split(' ')
      const description = article.description.split(' ')
      acc.push(...title, ...description)
    } else {
      const title = article.title.split(' ')
      acc.push(...title)
    }
    return acc;
  }, [])
  return words;
}

export const countWords = (arr) => {
  const countObj = arr.reduce((acc, word) => {
    const lowerWord = word.toLowerCase();
    if (!acc[lowerWord]) {
      acc[lowerWord] = 0
    }
    acc[lowerWord]++
    return acc;
  }, {})
  return countObj;
}

export const cleanWords = (arr) => {
  const commonWords = ['the', 'of', 'to', 'in', 'a', 'and', 'for', 'on', 'with', 'new', 'at', 'is', 'his', 'man', 'as', '-', '_', 'â€”', 'you', 'that', 'into', 'about', 'over', 'an', 'are', 'after', 'be', 'from', 'it', 'one', 'her', 'who', 'the', 'of', 'and', 'a', 'to', 'in', 'is', 'you', 'that', 'it', 'he', 'was', 'for', 'on', 'are', 'as', 'with', 'his', 'they', 'I', 'at', 'be', 'this', 'have', 'from', 'or', 'one', 'had', 'by', 'word', 'but', 'not', 'what', 'all', 'were', 'we', 'when', 'your', 'can', 'said', 'there', 'use', 'an', 'each', 'which', 'she', 'do', 'how', 'their', 'if', 'will', 'up', 'other', 'about', 'out', 'many', 'then', 'them', 'these', 'so', 'some', 'her', 'would', 'make', 'like', 'him', 'into', 'time', 'has', 'look', 'two', 'more', 'write', 'go', 'see', 'number', 'no', 'way', 'could', 'people', 'my', 'than', 'first', 'been', 'call', 'who', 'its', 'now', 'find', 'long', 'down', 'day', 'did', 'get', 'come', 'made', 'may', 'part', '--', '(ap)', 'says', 'news', 'went', 'times', 'fox']
  commonWords.forEach(commonWord => {
    arr.forEach(word => {
      if (word === commonWord) {
        arr.splice(arr.indexOf(word), 1)
      }
    })
  })
  return arr;
}

export const cleanSource = (source) => {
  if(source === 'Npr.org') {
    return 'NPR'
  } else if (source === 'Theatlantic.com') {
    return 'The Atlantic'
  } else if (source === 'Ap.org') {
    return 'The Associated Press'
  } else if (source === 'C-span.org') {
    return 'C-SPAN'
  } else if (source === 'Cato.org') {
    return 'The Cato Institute'
  } else {
    return source
  }
}

export const createMatchObject = (keywords, articles) => ({
  keywords,
  articles,
  id: Date.now()
})

export const flattenArrays = (arr) => {
  return arr.reduce((acc, arr) => {
    return acc.concat(Array.isArray(arr) ? flattenArrays(arr) : arr);
  }, []);
}

export const cleanArticles = (articles) => {
  const cleanedArticles = articles.map(article => {
    let cleanedArticle = {}
    if (!article.source) {
      cleanedArticle.source = {name: 'Unable to find article source'}
    } else {
      cleanedArticle.source = article.source
    }
    if (!article.title) {
      cleanedArticle.title = 'Unable to find article title'
    } else {
      cleanedArticle.title = article.title
    }
    if (!article.description) {
      cleanedArticle.description = 'Unable to find article description'
    } else {
      cleanedArticle.description = article.description
    }
    if (!article.url) {
      cleanedArticle.url = 'Unable to find link to article'
    } else {
      cleanedArticle.url = article.url
    }
    if (!article.urlToImage) {
      cleanedArticle.urlToImage = null
    }
    else {
      cleanedArticle.urlToImage = article.urlToImage
    }
    return cleanedArticle
  })
  return cleanedArticles
} 

export const cleanImageUrl = (url) => {
  if(!url) {
    return false;
  }
  const split = url.split('/')
  if(split[0] === 'https:' || split[0] === 'http:'){
    return true
  } else {
    return false
  }
}

export const topTen = (arr) => {
  const words = helper.makeWordsArr(arr)
  const wordObj = helper.countWords(words);
  const keys = Object.keys(wordObj);
  const sorted = keys.sort((a, b) => {
    return wordObj[b] - wordObj[a];
  })
  const cleaned = helper.cleanWords(sorted)
  return cleaned.slice(0, 10)
}