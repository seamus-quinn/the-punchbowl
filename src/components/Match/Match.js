import React from 'react';
import './Match.css';

const Match = (props) => {
  const { match } = props
  const keywords = match.keywords.map(keyword => {
    return(
      <h1>{keyword}</h1>
    )
  })
  const articles = match.articles.map(article => {
    return (
      <div className='article-card'>
        <img src={article.urlToImage} className='image' />
        <div className='card-info'>
          <h1 className='article-title'>
            {article.title}
          </h1>
          <p className='article-description'>
            {article.description}
          </p>
          <a
            href={article.url}
            className='link-to-article'
          >
            View Article
          </a>
        </div>
      </div>
    )
  })
  return (
    <div className='track'>
      <div className='keyword-card'>
        {keywords}
      </div>
      <div className='article-container'>
        {articles}
      </div>    
    </div>
  )
}

export default Match