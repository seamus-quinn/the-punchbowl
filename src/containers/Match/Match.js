import React from 'react';
import { connect } from 'react-redux'
import './Match.css';
import { deleteMatch } from '../../actions';
import * as helper from '../../helper';
import mitt from '../../assets/mitt.jpg'
import x from '../../assets/clear-button.svg'

const Match = (props) => {
  const { match } = props
  const keywords = match.keywords.map(keyword => {
    return(
      <h1>{keyword}</h1>
    )
  })

  const articles = match.articles.map(article => {
    const source = helper.cleanSource(article.source.name);
    const imageSource = helper.cleanImageUrl(article.urlToImage)

    return (
      <div className='article-card'>
        <div className='image-container'>
          <img src={!imageSource ? mitt : article.urlToImage} alt='' className='image' />
        </div>
        <div className='card-info'>
          <h1 className='article-title'>
            {article.title}
          </h1>
          <p className='article-description'>
            {article.description}
          </p>
          <p className='article-source'>Source: {source}</p>
          <a
            href={article.url}
            target='_blank'
            className='link-to-article'
          >
            View Article
          </a>
        </div>
      </div>
    )
  })
  return (
    <div className='track'>
      <div className='keyword-card'>
        <div className='keyword-text'>
          <h1>Matches:</h1>
          <div className='keyword-list'>
            {keywords}
          </div>
        </div>
        <button
          className='match-delete-button'
          onClick={() => props.deleteMatch(match.id)}
        >
          <img src={x} alt='' className='x' />
        </button>
        <p className='article-counter'>Number of Articles: {articles.length}</p>
      </div>
      <div className='article-container'>
        {articles}
      </div>    
    </div>
  )
}

const mapDispatchToProps = (dispatch) => ({
  deleteMatch: (match) => dispatch(deleteMatch(match))
})

export default connect(null, mapDispatchToProps)(Match)