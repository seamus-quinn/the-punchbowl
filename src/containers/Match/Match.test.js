import React from 'react'
import { Match}  from './Match'
import { shallow, mount } from 'enzyme';
import * as helper from '../../helper'

describe('Match', () => {

  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      keywords: ['garbage', 'mullet', 'sushi'],
      articles: [
        {
          title: 'Garbage Man eats Sushi',
          description: 'Wow much mullets worn by enthusiasts',
        }
      ],
      source: {
        name: 'I am Justice Beavers'
      }
    }
    wrapper = shallow(<Match match={mockProps} />)
  })

  it('matches the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  })

  it('calls helper function cleanSource with the correct arguments', () => {
    const wrapper = mount(<Match match={mockProps} />)
    helper.cleanSource = jest.fn();

    expect(helper.cleanSource).toHaveBeenCalledWith(mockProps.source.name)
  })
})