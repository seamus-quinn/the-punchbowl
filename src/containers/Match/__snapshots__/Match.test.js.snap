// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Match
    match={
        Object {
            "articles": Array [
              Object {
                "description": "Wow much mullets worn by enthusiasts",
                "title": "Garbage Man eats Sushi",
              },
            ],
            "keywords": Array [
              "garbage",
              "mullet",
              "sushi",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h1>
                    garbage
          </h1>
          <h1>
                    mullet
          </h1>
          <h1>
                    sushi
          </h1>
</div>,
        <div>
          <div>
                    <h1>
                              Garbage Man eats Sushi
                    </h1>
                    <p>
                              Wow much mullets worn by enthusiasts
                    </p>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              garbage
</h1>,
            <h1>
              mullet
</h1>,
            <h1>
              sushi
</h1>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "garbage",
            },
            "ref": null,
            "rendered": "garbage",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mullet",
            },
            "ref": null,
            "rendered": "mullet",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "sushi",
            },
            "ref": null,
            "rendered": "sushi",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h1>
                            Garbage Man eats Sushi
              </h1>
              <p>
                            Wow much mullets worn by enthusiasts
              </p>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Garbage Man eats Sushi
</h1>,
                <p>
                  Wow much mullets worn by enthusiasts
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Garbage Man eats Sushi",
                },
                "ref": null,
                "rendered": "Garbage Man eats Sushi",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wow much mullets worn by enthusiasts",
                },
                "ref": null,
                "rendered": "Wow much mullets worn by enthusiasts",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h1>
                        garbage
            </h1>
            <h1>
                        mullet
            </h1>
            <h1>
                        sushi
            </h1>
</div>,
          <div>
            <div>
                        <h1>
                                    Garbage Man eats Sushi
                        </h1>
                        <p>
                                    Wow much mullets worn by enthusiasts
                        </p>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                garbage
</h1>,
              <h1>
                mullet
</h1>,
              <h1>
                sushi
</h1>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "garbage",
              },
              "ref": null,
              "rendered": "garbage",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mullet",
              },
              "ref": null,
              "rendered": "mullet",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sushi",
              },
              "ref": null,
              "rendered": "sushi",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h1>
                                Garbage Man eats Sushi
                </h1>
                <p>
                                Wow much mullets worn by enthusiasts
                </p>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Garbage Man eats Sushi
</h1>,
                  <p>
                    Wow much mullets worn by enthusiasts
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Garbage Man eats Sushi",
                  },
                  "ref": null,
                  "rendered": "Garbage Man eats Sushi",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wow much mullets worn by enthusiasts",
                  },
                  "ref": null,
                  "rendered": "Wow much mullets worn by enthusiasts",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
